src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -682,6 +735,41 @@ void DlgPrefSound::settingChanged() {
         return; // doesn't count if we're just loading prefs
     }
     m_settingsModified = true;
+
+#ifdef __RUBBERBAND__
+    bool supportedScaler = keylockComboBox->currentData()
+                                   .value<EngineBuffer::KeylockEngine>() !=
+            EngineBuffer::KeylockEngine::SoundTouch;
+    bool monoMix = mainOutputModeComboBox->currentIndex() == 1;
+    keylockMultithreadedComboBox->setEnabled(!monoMix && supportedScaler);
+    keylockMultithreadedComboBox->setToolTip(monoMix
+                    ? kKeylockMultiThreadedUnavailableMono
+                    : (supportedScaler
+                                      ? kKeylockMultiThreadedAvailable
+                                      : kKeylockMultiThreadedUnavailableRubberband));
+}
+
+void DlgPrefSound::updateKeylockMultithreading(bool enabled) {
+    m_settingsModified = true;
+    if (!enabled) {
+        return;
+    }
+    QMessageBox msg;
+    msg.setIcon(QMessageBox::Warning);
+    msg.setWindowTitle(tr("Are you sure?"));
+    msg.setText(
+            tr("<p>Using multi threading may result in <span "
+               "style=\"font-weight:600;\">pitch and tone imperfection</span> "
+               "depending of the platform, leading to mono-incompatibility, "
+               "due to third party limitations. </p><p>Are you sure you wish "
+               "to proceed?</p>"));
```

Comment:
Is this true? I am not aware of a pitch and tone imperfection in the single channel case. I have just found this: 
https://breakfastquay.com/rubberband/code-doc/classRubberBand_1_1RubberBandStretcher.html

> OptionChannelsApart - Channels are handled for maximum individual fidelity, at the expense of synchronisation. In the R3 engine, this means frequency-bin synchronisation is maintained more closely for lower-frequency content than higher. In R2, it means the stereo channels are processed individually and only synchronised at transients. In both engines this gives the highest quality for the individual channels but a more diffuse stereo image, an unnatural increase in "width", and generally a loss of mono compatibility (i.e. mono mixes from stereo can sound phasy). This option is the default.

> OptionChannelsTogether - Channels are handled for higher synchronisation at some expense of individual fidelity. In particular, a stretcher processing two channels will treat its input as a stereo pair and aim to maximise clarity at the centre and preserve mono compatibility. This gives relatively less stereo space and width than the default, as well as slightly lower fidelity for individual channel content, but the results may be more appropriate for many situations making use of stereo mixes.

So how about this warning:
```
Using multi threading result in a loss of mono compatibility and a diffuse stereo image. It is not recommended during broadcasting or recording. 
```

Reply:
That is much better, thank you.